sqlx-postgres/src/types/geometry/path.rs

```
@@ -0,0 +1,334 @@
+/// ```
+/// where the points are the end points of the line segments comprising the path. Square brackets `([])` indicate an open path, while parentheses `(())` indicate a closed path.
+/// When the outermost parentheses are omitted, as in the third through fifth syntaxes, a closed path is assumed.
+///
+/// Paths are output using the first or second syntax, as appropriate.
+///
+/// See https://www.postgresql.org/docs/16/datatype-geometric.html#DATATYPE-GEOMETRIC-PATHS
+#[derive(Debug, Clone, PartialEq)]
+pub struct PgPath {
+    pub closed: bool,
+    pub points: Vec<PgPoint>,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+struct Header {
+    is_closed: bool,
+    length: usize,
+}
+
+impl Type<Postgres> for PgPath {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::with_name("path")
+    }
+}
+
+impl PgHasArrayType for PgPath {
+    fn array_type_info() -> PgTypeInfo {
+        PgTypeInfo::with_name("_path")
+    }
+}
+
+impl<'r> Decode<'r, Postgres> for PgPath {
+    fn decode(value: PgValueRef<'r>) -> Result<Self, Box<dyn std::error::Error + Send + Sync>> {
+        match value.format() {
+            PgValueFormat::Text => Ok(PgPath::from_str(value.as_str()?)?),
+            PgValueFormat::Binary => Ok(PgPath::from_bytes(value.as_bytes()?)?),
+        }
+    }
+}
+
+impl<'q> Encode<'q, Postgres> for PgPath {
+    fn produces(&self) -> Option<PgTypeInfo> {
+        Some(PgTypeInfo::with_name("path"))
+    }
+
+    fn encode_by_ref(&self, buf: &mut PgArgumentBuffer) -> Result<IsNull, BoxDynError> {
+        self.serialize(buf)?;
+        Ok(IsNull::No)
+    }
+}
+
+impl FromStr for PgPath {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let closed = !s.contains("[");
+        let sanitised = s.replace(&['(', ')', '[', ']', ' '][..], "");
+        let mut parts = sanitised.split(",");
+
+        let mut points = vec![];
+
+        while let (Some(x_str), Some(y_str)) = (parts.next(), parts.next()) {
```

This silently drops the last coordinate for uneven numbers.
=> Please make that an error instead